### PR TITLE
[PHP 8.0] Improve NullsafeOperatorRector : Check against !== null

### PR DESCRIPTION
**Pull request by samsonasik**
Added ability for : 

```diff
-        if ($o !== null) {
-            $user = $o->user;
-            if ($user !== null) {
-                $address = $user->getAddress();
-                if ($address !== null) {
-                    $country = $address->getCountry();
-                }
-            }
-        }
+        $country = $o?->user?->getAddress()?->getCountry();
```

### CODE REVIEW

**Comment by samsonasik** [comment]
I added yoda condition check as well.

File: phpstan.neon [hunk 1]
```diff
@@ -692,6 +692,7 @@ parameters:
         - '#Class with base "Application" name is already used in "(.*?)\\Symfony\\Component\\Console\\Application", "Symfony\\Component\\Console\\Application", "Rector\\Core\\Console\\Application"\. Use unique name to make classes easy to recognize#'
         - '#Class with base "FileNode" name is already used in "PHPStan\\Node\\FileNode", "Rector\\Core\\PhpParser\\Node\\CustomNode\\FileNode"\. Use unique name to make classes easy to recognize#'
         - '#Class cognitive complexity for "BetterStandardPrinter" is 52, keep it under 50#'
+        - '#Class cognitive complexity for "NullsafeOperatorRector" is 53, keep it under 50#'
```
Comment (TomasVotruba) [inline phpstan.neon:695]: These should not be ignored, but rather extract a new service with some common logic

Comment (samsonasik) [reply phpstan.neon]: I will try

Comment (samsonasik) [reply phpstan.neon]: implemented, moved check if cond using assign identical/not identical to IfManipulator service

**Comment by samsonasik** [comment]
All checks have passed :tada: @TomasVotruba it is ready for review.

File: phpstan.neon (hunk 1 above)
Comment (samsonasik) [reply phpstan.neon]: I moved NullSafe create process to new service named `NullsafeManipulator`

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: src/PhpParser/Node/Manipulator/NullsafeManipulator.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+final class NullsafeManipulator
+{
+    public function processNullSafeExpr(Expr $expr): ?Expr
+    {
+        if ($expr instanceof MethodCall) {
+            return new NullsafeMethodCall($expr->var, $expr->name);
+        }
+
+        if (property_exists($expr, 'var') && property_exists($expr, 'name')) {
```
Comment (TomasVotruba) [inline src/PhpParser/Node/Manipulator/NullsafeManipulator.php:22]: This is way too magical check. AST cannot analyse this (PHPStan, Rector and PHPStorm will fail on changed nodes).

It should be handled by `$node instanceof <specificTypeNode>` instead

File: rules/php80/src/Rector/If_/NullsafeOperatorRector.php [hunk 3]
```diff
@@ -37,9 +38,15 @@ final class NullsafeOperatorRector extends AbstractRector
      */
     private $ifManipulator;
 
-    public function __construct(IfManipulator $ifManipulator)
+    /**
+     * @var NullsafeManipulator
+     */
+    private $nullsafeManipulator;
```
Comment (TomasVotruba) [inline rules/php80/src/Rector/If_/NullsafeOperatorRector.php:43]: :+1:

File: src/PhpParser/Node/Manipulator/NullsafeManipulator.php (hunk 2 above)
Comment (samsonasik) [reply src/PhpParser/Node/Manipulator/NullsafeManipulator.php]: implemented.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you

**Pull request merged**